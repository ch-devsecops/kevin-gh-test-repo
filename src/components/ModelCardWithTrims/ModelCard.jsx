import React from 'react';
import { Box, Image, H5, H6, Copy, Optional } from '@honda-canada/design-system-react';
import { getGtmTagValue } from '../../utils/gtmEvents';
import CTA from '../CTA';
import themeStyles from './ModelCardWithTrims.styles';

const Container = themeStyles.apply(Box, 'ModelCardWrapper');
const GradientContainer = themeStyles.apply(Box, 'ModelCardGradientContainer');
const Content = themeStyles.apply(Box, 'ModelCardContent');
const H5ToggleButton = themeStyles.apply(H5, 'ModelCardTaglineComponent');
const H6ToggleButton = themeStyles.apply(H6, 'ModelCardTaglineComponent');
const Description = themeStyles.apply(Copy, 'ModelCardDescription');

const headingMap = {
  h5: H5ToggleButton,
  h6: H6ToggleButton,
};

const ModelCard = ({
  tagline,
  description,
  badgeImageSrc,
  badgeImageAlt,
  backgroundImageSrc,
  exploreCta,
  showModelCardImage,
  modelCardTaglineComponent,
  modelCardDescriptionLineHeight,
  modelCardDescriptionFontSize,
  componentName,
  gtmCategory,
}) => {
  const TaglineComponent = headingMap[modelCardTaglineComponent];
  return (
    <Container backgroundImage={backgroundImageSrc}>
      <GradientContainer>
        <Content>
          <Optional when={showModelCardImage}>
            <Image src={badgeImageSrc} alt={badgeImageAlt} width={['110px', '160.27px']} mb="xs" />
          </Optional>
          <TaglineComponent color="white" data-testid="title">
            {tagline}
          </TaglineComponent>
          <Optional when={description}>
            <Description
              lineHeight={modelCardDescriptionLineHeight}
              fontSize={modelCardDescriptionFontSize}
              data-testid="description"
            >
              {description}
            </Description>
          </Optional>
          <Optional when={exploreCta?.ctaLink?.value?.href && exploreCta?.ctaLink?.value?.text}>
            <CTA
              data-testid="cy-model-card-cta"
              linkField={exploreCta.ctaLink}
              typeField={{ value: 'TertiaryWhite' }}
              iconColor="white"
              icon="arrowRight"
              parentGtmTags={{
                'data-gtm-component-type': getGtmTagValue(componentName),
                'data-gtm-category': getGtmTagValue(gtmCategory),
              }}
              gtmTags={{
                'data-gtm-interaction-type': exploreCta.gtmTags?.gtmInteractionType?.value,
                'data-gtm-model': exploreCta.gtmTags?.gtmModelName,
                'data-gtm-body-style': exploreCta.gtmTags?.gtmBodyStyle,
                'data-gtm-title': 'vehicle tile',
              }}
              ariaLabel={`${exploreCta.ctaLink.value.text} ${exploreCta.gtmTags?.gtmModelName}`}
            />
          </Optional>
        </Content>
      </GradientContainer>
    </Container>
  );
};

export default ModelCard;
