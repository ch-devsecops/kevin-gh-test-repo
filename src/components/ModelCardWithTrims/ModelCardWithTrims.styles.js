import { ThemeStyles } from '@honda-canada/design-system-react';
import css from '@styled-system/css';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('AlertContainer')(
  css({
    justifyContent: 'center',
    my: ['18px', '136px', '136px'],
    mx: ['56px', '0', '0'],
  }),
);

themeStyles.addBaseStyles('AlertMarkdown')(
  css({
    lineHeight: ['24px', '24px', '28px'],
    textAlign: 'center',
    color: 'black',
  }),
);

themeStyles.addBaseStyles('ProductCardWrapper')(
  css({
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    pt: 'default',
    height: '100%',
  }),
);

themeStyles.addBaseStyles('ProductCardRow')(
  css({
    pt: 'default',
  }),
);

themeStyles.addBaseStyles('ProductCardCol')(
  css({
    mb: 'm',
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
  }),
);

themeStyles.addBaseStyles('TrimCardMobileWrapper')(
  css({
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    pt: 'default',
  }),
);

themeStyles.addBaseStyles('TrimCardDesktopWrapper')(({ trimCardAlignment }) =>
  css({
    display: 'flex',
    justifyContent: trimCardAlignment,
    flexWrap: 'wrap',
    pt: 'default',
  }),
);

themeStyles.addBaseStyles('TrimCardRow')(
  css({
    display: 'flex',
    justifyContent: 'center',
    flexWrap: 'wrap',
    pt: 'default',
  }),
);

themeStyles.addBaseStyles('TrimCardColumn')(
  css({
    width: ['100%', '50%', '33%'],
    mb: 'm',
    alignItems: 'center',
    display: ['flex', 'flex', 'block'],
    flexDirection: 'column',
  }),
);

themeStyles.addBaseStyles('CTAsWrapper')(
  css({
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '285px',
    minHeight: '80px',
    alignItems: 'center',
    mx: 'auto',
  }),
);

themeStyles.addBaseStyles('ModelsContainer')(({ isExpanded }) =>
  css({
    mb: isExpanded ? 'xl' : 'zero',
    display: isExpanded ? 'block' : 'none',
  }),
);

themeStyles.addBaseStyles('ExpandButtonWrapper')(
  css({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    mx: 'auto',
    py: 'm',
  }),
);

themeStyles.addBaseStyles('ExpandButtonLabel')(
  css({
    fontFamily: 'bold',
    size: 'regular',
  }),
);

themeStyles.addBaseStyles('StyledExpandButton')(({ theme }) =>
  css({
    display: 'flex',
    border: 'none',
    alignItems: 'center',
    '&:hover, &:focus': {
      border: 'tertiaryLight',
    },
    '&:focus-visible': {
      outline: `${theme.borders[8]} ${theme.colors.aodaFocused}`,
    },

    '> p': {
      fontSize: '14px !important',
    },
  }),
);

themeStyles.addBaseStyles('ModelCardWrapper')(
  css({
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    height: ['240px', '350px', '350px'],
    backgroundPosition: ['center top', 'center top', 'center'],
  }),
);

themeStyles.addBaseStyles('ModelCardGradientContainer')(
  css({
    width: '100%',
    height: '100%',
    background: [
      'linear-gradient(0deg,#000 10%, rgba(0,0,0,0.1), rgba(0,0,0,0))',
      'radial-gradient(at bottom left,#000,rgba(0,0,0,0),rgba(0,0,0,0))',
    ],
    display: 'flex',
    alignItems: 'flex-end',
    px: ['20px', 6, 6],
    py: [4, 5, 5],
  }),
);

themeStyles.addBaseStyles('ModelCardContent')(
  css({
    display: 'flex',
    flexDirection: 'column',
  }),
);

themeStyles.addBaseStyles('ModelCardTaglineComponent')(
  css({
    mb: ['xs', 'm', 'm'],
  }),
);

themeStyles.addBaseStyles('ModelCardDescription')(
  css({
    color: 'white',
    mb: ['xs', 'm', 'm'],
    maxWidth: '360px',
  }),
);

themeStyles.addBaseStyles('ExteriorDisclaimerContainer')(({ exteriorDisclaimerMarginTop }) =>
  css({
    mt: exteriorDisclaimerMarginTop,
    mx: ['xxl', 'auto'],
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    color: 'blue',
  }),
);

themeStyles.addBaseStyles('ExteriorDisclaimerLabel')(
  css({
    fontFamily: 'bold',
    size: 'small',
    color: 'tertiaryLight',
  }),
);

themeStyles.addBaseStyles('ExteriorIconContainer')(
  css({
    size: 'iconWrapper.sm',
    height: 'auto',
    alignItems: 'flex-start',
  }),
);

export default themeStyles;
