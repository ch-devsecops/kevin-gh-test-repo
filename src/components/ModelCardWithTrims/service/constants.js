export const variant1 = 'variant1';
export const variant2 = 'variant2';
export const variant3 = 'variant3';
export const variant4 = 'variant4';

export const PAYMENT_METHOD = 'finance';
export const PAYMENT_FREQUENCY = 'weekly';

/**
 * This empty data structure serves as a baseline for the modelCardWithTrims component.
 * Note that all the properties are generically named and matches between Marine and Auto brands.
 * However we are left with Trims and Transmissions that sound auto specific
 * and they map to Models and Configurations respectively.
 */

export const emptyDataFields = {
  data: {
    value: {
      heroImage: {},
      useDefaultModelYear: { value: '' },
      paymentMethod: { value: '' },
      paymentFrequency: { value: '' },
      showInformationalApr: { value: '' },
      model: {},
    },
  },
};

export const emptyDataModel = {
  fields: {
    defaultYear: { fields: {} },
    nameBadge: { item: { value: {} } },
    sotId: { value: '' },
  },
};

export const emptyModelYears = {
  defaultTrim: { fields: {} },
  detKey: { value: '' },
  emissionSegmentAverage: { value: '' },
  legacyPageUrl: { value: '' },
  sotId: { value: '' },
  tagline: { value: '' },
  trimLegacyPageUrl: { value: '' },
  year: { value: '' },
};

export const emptyDataTrim = {
  defaultTransmission: { item: {} },
  detKey: { value: '' },
  emissionRating: { value: '' },
  nameBadge: { item: { value: {} } },
  primaryThumbnail: { item: { value: {} } },
  secondaryThumbnail: { item: { value: {} } },
  sotId: { value: '' },
  specialVehicleType: { item: null },
  trimName: { value: '' },
};
