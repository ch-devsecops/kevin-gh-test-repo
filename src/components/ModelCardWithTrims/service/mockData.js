export const Model = {
  NameBadge: '/data/media/img/model-card-with-trims/badge.svg',
  Year: {
    Tagline: 'Compact Sport Sedan',
  },
  ModelKey: 'rdx',
  ModelYear: 2020,
  Trims: [
    {
      TrimName: 'trim name',
      TrimKey: 'rdx_10226',
      TrimThumbnails: {
        PrimaryThumbnail: {
          src: '/data/media/img/model-card-with-trims/primary-thumbnail.png',
          alt: 'Primary thumbnail',
        },
      },
      NameBadge: '/data/media/img/model-card-with-trims/trim-badge.svg',
      DefaultTransmision: '10201-Automatic',
      Transmissions: [
        {
          Name: 'Transmission A',
          TransmissionModelCode: '10201-Automatic',
          IsBuildable: true,
          HasSpecs: true,
          InteriorColorKey: 'kkebony_leatherette_^my20_ilx',
          ExteriorColorKey: 'bkebony_leatherette_^my20_rdx',
        },
        {
          Name: 'Transmission B',
          TransmissionModelCode: 'transmission-b',
          IsBuildable: true,
          HasSpecs: false,
        },
        {
          Name: 'Transmission C',
          TransmissionModelCode: 'transmission-c',
          IsBuildable: false,
          HasSpecs: true,
        },
        {
          Name: 'Transmission D',
          TransmissionModelCode: 'transmission-d',
          IsBuildable: false,
          HasSpecs: false,
        },
      ],
    },
  ],
};

export const strings = {
  Shared: {
    ModelCardWithTrims: {
      exploreModelLabel: 'Explore Model',
      msrpStartingFromLabel: 'MSRP Starting From*',
      offersAvailableLabel: 'Offers Available',
      buildAndPriceLabel: 'Build & Price',
      viewTrimSpecsLabel: 'View Trim Specs',
      priceErrorLabel: 'Unable to load pricing. Please reload the page',
    },
  },
};

export default { Model, strings };
