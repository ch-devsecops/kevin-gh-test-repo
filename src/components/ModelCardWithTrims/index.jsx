import React from 'react';
import PropTypes from 'prop-types';

import ModelCardWithTrims from './ModelCardWithTrims';
import { ACURA_SITE_NAME, HONDA_SITE_NAME, MARINE_SITE_NAME, PSP_SITE_NAME } from '../../utils/constants';
import useAppName from '../../utils/sitecoreContext/useAppName';
import { isEmpty } from '../../utils/object';
import { mapMarineFields, mapPSPFields } from './service/utils';
import { variant1, variant2, variant3, variant4 } from './service/constants';

const ModelCardWithTrimsWrapper = ({ variant: propVariant, fields, rendering, priceLabels }) => {
  const appName = useAppName();

  if (!fields || isEmpty(fields)) return null;

  let variant;
  let vehicleType;
  let mappedFields = fields;

  switch (appName) {
    case PSP_SITE_NAME:
      variant = variant4;
      // Maps from PSP Category/Product => Model/Trim
      mappedFields = mapPSPFields(mappedFields);
      vehicleType = mappedFields?.data?.value?.category?.fields?.vehicleType?.name;

      break;

    case MARINE_SITE_NAME:
      variant = variant3;
      // Maps from Marine Category/Product => Model/Trim
      mappedFields = mapMarineFields(mappedFields);
      break;

    case ACURA_SITE_NAME:
      variant = variant2;

      break;

    case HONDA_SITE_NAME:
    default:
      variant = variant1;
      break;
  }

  if (mappedFields === null || isEmpty(mappedFields)) {
    return null;
  }

  return (
    <ModelCardWithTrims
      componentName={rendering?.componentName}
      fields={mappedFields}
      priceLabels={priceLabels}
      variant={propVariant ?? variant}
      vehicleType={vehicleType}
    />
  );
};

ModelCardWithTrimsWrapper.propTypes = {
  fields: PropTypes.shape({
    data: PropTypes.shape({
      value: PropTypes.shape({
        engineSeries: PropTypes.shape({
          fields: PropTypes.shape({
            name: PropTypes.string,
          }),
        }),
      }),
    }),
  }),
  variant: PropTypes.string,
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};

export default ModelCardWithTrimsWrapper;
