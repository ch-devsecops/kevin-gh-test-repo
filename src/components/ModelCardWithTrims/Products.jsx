import React, { useContext } from 'react';
import { Box, Column, Media, Row } from '@honda-canada/design-system-react';
import { SplideSlide } from '@splidejs/react-splide';

import ProductCard from '../ProductCard';
import CarouselSlider from '../CarouselSlider';
import { ModelFiltersContext } from '../ModelFiltersContext';
import { getModelProduct } from '../ModelFiltersContext/utils';
import { EqualHeight } from '../../utils/components/EqualHeight';

import { usePricesFromFinancialModel } from '../../utils/financeUtils';
import themeStyles from './ModelCardWithTrims.styles';

const ProductCardWrapper = themeStyles.apply(Box, 'ProductCardWrapper');

const Products = ({ models, gtmTags, priceLabels }) => {
  const { getPrices } = usePricesFromFinancialModel(priceLabels, 'models');
  const modelFiltersContext = useContext(ModelFiltersContext);
  const { financials } = modelFiltersContext || {};

  const products = models?.map(model => ({ ...model, key: model?.key || model?.detKey?.value }));

  return (
    <>
      <Media at="mobile">
        <EqualHeight>
          <CarouselSlider length={products?.length} splideOptions={{ padding: { left: '0vw', right: '0vw' } }}>
            {() =>
              products?.map(model => {
                const prices = getPrices(getModelProduct(financials?.models, model?.detModelKey?.value));
                return (
                  <SplideSlide key={model.key}>
                    <ProductCardWrapper>
                      <ProductCard
                        prices={prices}
                        fields={model}
                        gtmTags={{ ...gtmTags, model: model?.modelName?.value }}
                      />
                    </ProductCardWrapper>
                  </SplideSlide>
                );
              })
            }
          </CarouselSlider>
        </EqualHeight>
      </Media>
      <Media greaterThanOrEqual="smallDesktop">
        <Row as="ul" pt="default">
          <EqualHeight>
            {products?.map(model => {
              const prices = getPrices(getModelProduct(financials?.models, model?.detModelKey?.value));
              return (
                <Column
                  key={model.key}
                  width={[1, 1 / 2, 1 / 3]}
                  mb="m"
                  as="li"
                  alignItems="center"
                  display="flex"
                  flexDirection="column"
                >
                  <ProductCard
                    prices={prices}
                    fields={model}
                    gtmTags={{ ...gtmTags, model: model?.modelName?.value }}
                  />
                </Column>
              );
            })}
          </EqualHeight>
        </Row>
      </Media>
    </>
  );
};

export default Products;
