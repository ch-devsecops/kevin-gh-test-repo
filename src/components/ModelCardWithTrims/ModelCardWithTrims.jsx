import React, { useEffect, useContext, useState, useMemo } from 'react';
import { Fade, Box, Optional } from '@honda-canada/design-system-react';
import { InView } from 'react-intersection-observer';
import get from 'lodash/get';
import uniqBy from 'lodash/uniqBy';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import { ModelFiltersContext } from '../ModelFiltersContext';
import { getFilteredModels } from '../ModelFiltersContext/utils';

import ModelCard from './ModelCard';
import Trims from './Trims';
import Products from './Products';
import ExpandButton from './shared/ExpandButton';
import ExteriorDisclaimer from './shared/ExteriorDisclaimer';
import { parseActiveModel } from './service/utils';

import { mapGTMCategory } from '../../utils/sitecoreFields';
import usePriceFlagsForProvince from '../PriceComponent/utils';
import NoFilterResult from './shared/NoFilterResult';
import themeStyles from './ModelCardWithTrims.styles';
import Context from './service/Context';
import { isEmpty } from '../../utils/object';
import { MODEL_YEAR_API_DEFAULT_VALUE } from '../../utils/constants';
import safelyStringifyJSON from '../../utils/safelyStringifyJSON';
import ConfigurationProvider from './service/Provider';

const ModelsContainer = themeStyles.apply(Box, 'ModelsContainer');

const productCardMap = {
  trims: Trims,
  products: Products,
};

const ModelCardWithTrimsUI = ({ fields, componentName }) => {
  const {
    collapseIfFilterChanged,
    hondaExpandProps,
    isExpandedDefault,
    modelCardDescriptionPath,
    modelCardProps,
    priceDetKey,
    priceLabels,
    priceModelKey,
    productsComponent,
    showExpandButton,
    vehicleType,
  } = useContext(Context);
  const ProductsComponent = productCardMap[productsComponent];

  const { showMsrpPrice, showSellingPrice } = usePriceFlagsForProvince();
  const { t } = useTranslation();
  const [isExpanded, setIsExpanded] = useState(isExpandedDefault);
  const modelFiltersContext = useContext(ModelFiltersContext);
  const {
    filter = {},
    setTrimCount,
    modelPrices = {},
    trimCount,
    setModelKeys,
    setVehicleType,
  } = modelFiltersContext || {};
  const categoryName = fields?.data?.value?.model?.fields?.detKey?.value;
  const category = fields?.data?.value?.model?.fields?.name;
  useEffect(() => {
    if (vehicleType) {
      setVehicleType(vehicleType);
    }
  }, []);

  const isFiltered = Object.keys(filter?.values || {})
    .map(filterKey => filter?.values[filterKey])
    .some(filterValue => filterValue?.length > 0);
  const isExteriorColorFiltered = Boolean(filter?.values?.exteriorColor.length);
  const { modelKey, tagline, badgeImage, heroImage, allModelYearTrims, exploreCta } = parseActiveModel(fields?.data);
  const paymentOptions = {
    paymentMethod: fields?.data?.value?.paymentMethod?.value?.toLowerCase() || 'finance',
    paymentFrequency: fields?.data?.value?.paymentFrequency?.value?.toLowerCase() || 'weekly',
  };

  // This function maps prices to models
  const modelsWithPrice = useMemo(() => {
    const modelKeys = Object.keys(modelPrices);
    const mappedPriceToModels = allModelYearTrims?.map(model => {
      if (modelKeys.indexOf(get(model, priceModelKey)) !== -1) {
        // eslint-disable-next-line no-param-reassign
        model.price = modelPrices[get(model, priceModelKey)];
      }
      return model;
    });
    return mappedPriceToModels || [];
  }, [safelyStringifyJSON(modelPrices)]);

  const validModelKeys =
    modelsWithPrice?.map(model => ({
      modelKey: get(model, priceDetKey),
      modelYear: model?.modelYear || MODEL_YEAR_API_DEFAULT_VALUE,
    })) || [];

  useEffect(() => {
    if (validModelKeys.length) {
      setModelKeys({ modelKey, modelIds: uniqBy(validModelKeys.filter(m => !!m.modelKey)) });
    }
  }, [safelyStringifyJSON(modelsWithPrice)]);

  const showInformationalApr = fields?.data?.value?.showInformationalApr?.value === '1';
  const filteredModels = getFilteredModels(
    modelsWithPrice,
    filter?.values,
    filter?.priceRange,
    showMsrpPrice,
    showSellingPrice,
  );

  const filteredModelsCount = filteredModels?.length;
  const filterValuesString = safelyStringifyJSON(filter?.values);
  const gtmCategory = fields?.data?.value?.gtmCategory;
  const gtmModelName = fields?.data?.value?.gtmModelName?.value;
  const gtmBodyStyle = fields?.data?.value?.gtmBodyStyle?.value;
  const productCardGtm = {
    componentName,
    bodyStyle: vehicleType?.toLowerCase(),
  };

  useEffect(() => {
    if ((!filter?.onlyFutureVehicles && !filter?.onlyNsx) || !filter) {
      setTrimCount({ modelKey, count: filteredModelsCount });
    }
    if (collapseIfFilterChanged) {
      setIsExpanded(false);
    }
    /* eslint-disable-next-line react-hooks/exhaustive-deps */
  }, [
    filterValuesString,
    filter?.onlyFutureVehicles,
    filter?.onlyNsx,
    !isEmpty(filter?.priceRange),
    filteredModelsCount,
  ]);

  // This component will show up only when the trim count is 0 for all categories

  if (trimCount === 0 && Object.keys(filter?.trimCounts)?.[0] === categoryName && !filter?.onlyFutureVehicles) {
    return <NoFilterResult content={t('Pages.Models.Exploration.filterNoResultsLabel')} />;
  }

  // This component is filterable. See ModelListFilters/reducer.js
  if (!fields || !fields?.data || filter?.onlyFutureVehicles || filter?.onlyNsx) {
    return null;
  }

  return (
    <InView threshold={0.2} triggerOnce>
      {({ inView, ref }) => (
        <div ref={ref}>
          <Optional when={filteredModels?.length > 0}>
            <Fade
              data-gtm-category={mapGTMCategory(gtmCategory)}
              data-gtm-component-type={componentName}
              shouldAnimate={isFiltered}
              initialOpacity={isFiltered ? 0 : 1}
              duration="t6"
              data-testid="ModelCardWithTrims"
            >
              <ModelCard
                tagline={tagline}
                description={get(fields, modelCardDescriptionPath)}
                exploreCta={exploreCta}
                backgroundImageSrc={heroImage}
                badgeImageSrc={badgeImage?.src}
                badgeImageAlt={badgeImage?.alt}
                componentName={componentName}
                gtmCategory={mapGTMCategory(gtmCategory)}
                {...modelCardProps}
              />
              <Optional when={showExpandButton}>
                <ExpandButton
                  category={category}
                  trimCount={filteredModelsCount}
                  isExpanded={isExpanded}
                  onClick={() => setIsExpanded(!isExpanded)}
                  gtmComponentType={componentName}
                  gtmModel={gtmModelName}
                  gtmBodyStyle={gtmBodyStyle}
                  gtmTitle={tagline}
                  gtmCategory={vehicleType || category?.toLowerCase()}
                  {...hondaExpandProps}
                />
              </Optional>
              <ModelsContainer isExpanded={isExpanded}>
                <Optional when={isExteriorColorFiltered}>
                  <ExteriorDisclaimer />
                </Optional>
                <Optional when={isExpanded}>
                  {/* Based on variants */}
                  <ProductsComponent
                    models={filteredModels}
                    componentName={componentName}
                    gtmTags={productCardGtm}
                    priceLabels={priceLabels}
                    paymentOptions={paymentOptions}
                    shouldFetch={inView && isExpanded}
                    showInformationalApr={showInformationalApr}
                  />
                </Optional>
              </ModelsContainer>
            </Fade>
          </Optional>
        </div>
      )}
    </InView>
  );
};

ModelCardWithTrimsUI.propTypes = {
  fields: PropTypes.shape({}),
  componentName: PropTypes.string,
};

const ModelCardWithTrims = ({ variant, vehicleType, priceLabels, ...restProps }) => (
  <ConfigurationProvider variant={variant} vehicleType={vehicleType} priceLabels={priceLabels}>
    <ModelCardWithTrimsUI {...restProps} />
  </ConfigurationProvider>
);

ModelCardWithTrims.propTypes = {
  componentName: PropTypes.string,
  fields: PropTypes.shape({}),
  priceLabels: PropTypes.shape({}),
  variant: PropTypes.string,
  vehicleType: PropTypes.string,
};

export default ModelCardWithTrims;
