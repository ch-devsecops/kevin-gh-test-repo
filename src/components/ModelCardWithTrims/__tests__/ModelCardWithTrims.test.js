import React from 'react';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { jssRender, cleanup, axe } from 'test-utils';

import ModelCardWithTrims from '../index';
import { variant3 } from '../service/constants';
import { MARINE_SITE_NAME } from '../../../utils/constants';

afterEach(cleanup);

const mockData = require('../../../../sample-mock-data/routes/marine-category-with-products/en.json');
const emptyMockData = require('../../../../sample-mock-data/routes/marine-category-with-products/empty_en.json');

describe('Component MarineCategoryWithProducts as ModelCardWithTrims for Marine', () => {
  const sitecoreContextMock = { site: { name: MARINE_SITE_NAME } };

  test('should render correctly', () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ModelCardWithTrims {...mockData} variant={variant3} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should render correctly with empty mock data', () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ModelCardWithTrims {...emptyMockData} variant={variant3} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
  });

  test('should return null if there is no  fields', () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ModelCardWithTrims fields={null} variant={variant3} />
      </SitecoreContext>,
    );

    expect(container).toMatchSnapshot();
  });

  test('should have no accessibility violations', async () => {
    const { container } = jssRender(
      <SitecoreContext context={sitecoreContextMock} componentFactory={{}}>
        <ModelCardWithTrims {...mockData} variant={variant3} />
      </SitecoreContext>,
    );

    expect(await axe(container)).toHaveNoViolations();
  });
});
