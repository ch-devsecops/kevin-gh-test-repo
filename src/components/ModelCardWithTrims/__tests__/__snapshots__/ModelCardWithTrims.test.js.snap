// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MarineCategoryWithProducts as ModelCardWithTrims for Marine should render correctly 1`] = `
<div>
  <div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 jEZHCt"
      data-gtm-category="marine gtm category"
      data-gtm-component-type="MarineCategoryWithProducts"
      data-testid="ModelCardWithTrims"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 fgBlzk"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 xLfJ"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 dnkuzx"
          >
            <h5
              class="sc-1k44dfv-0 sc-1k44dfv-0 eBsHxQ"
              color="white"
              data-testid="title"
            >
              Portables
            </h5>
            <p
              class="sc-1k44dfv-0 sc-1k44dfv-0 hsuEQC"
              color="typographyDefault"
              data-testid="description"
              font-family="default"
              font-size="14px"
              font-weight="regular"
            >
              Big on convenience, versatility and performance, Honda portable outboard motors are easy to use and easy to transport.
            </p>
            <span
              data-gtm-category="marine gtm category"
              data-gtm-component-type="MarineCategoryWithProducts"
            >
              <a
                aria-label="Explore Portables portables"
                class="Link-sc-16wd4a0-0 dKSZAU"
                data-gtm-body-style=""
                data-gtm-interaction-type=""
                data-gtm-model="portables"
                data-gtm-title="vehicle tile"
                data-testid="cy-model-card-cta"
                href="/en/portables"
                target=""
                title=""
              >
                Explore Portables
              </a>
            </span>
          </div>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gRkQXT"
        data-gtm-category="portables"
        data-gtm-component-type="MarineCategoryWithProducts"
        data-testid="expand-button-Portables"
      >
        <button
          class="Button-pulw6t-0 sc-1k44dfv-0 ikZvLZ"
          data-gtm-category="portables"
          data-gtm-interaction-type="cta: category page"
          data-gtm-model="marine gtm model"
          data-gtm-title="Portables"
          data-testid="expand-button"
        >
          <p
            class="sc-1k44dfv-0 sc-1k44dfv-0 KueHw"
            color="typographyDefault"
            data-testid="model-count"
            font-family="default"
            font-weight="regular"
          >
            4 Pages.Models.Exploration.modelsAvailableLabel
          </p>
          <div
            class="IconWrapper-lp9q2r-0 bbbwgU"
          />
        </button>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 hhKxgB"
      />
    </div>
  </div>
</div>
`;

exports[`Component MarineCategoryWithProducts as ModelCardWithTrims for Marine should render correctly with empty mock data 1`] = `<div />`;

exports[`Component MarineCategoryWithProducts as ModelCardWithTrims for Marine should return null if there is no  fields 1`] = `<div />`;
